mlflow/java/client/src/main/java/org/mlflow/client/ApiClient.java

```diff
@@ -0,0 +1,140 @@
+      .getExperimentsList();
+  }
+
+  public long createExperiment(String experimentName) throws Exception {
+    String ijson = fromMapper.makeCreateExperimentRequest(experimentName);
+    String ojson = httpCaller.post("experiments/create", ijson);
+    return toMapper.toCreateExperimentResponse(ojson).getExperimentId();
+  }
+
+  public Run getRun(String runUuid) throws Exception {
+    URIBuilder builder = new URIBuilder("runs/get").setParameter("run_uuid", runUuid);
+    return toMapper.toGetRunResponse(httpCaller.get(builder.toString())).getRun();
+  }
+
+  public RunInfo createRun(CreateRun request) throws Exception {
+    String ijson = fromMapper.toJson(request);
+    String ojson = post("runs/create", ijson);
+    return toMapper.toCreateRunResponse(ojson).getRun().getInfo();
+  }
+
+  public RunInfo createRun() throws Exception {
+    return createRun(CreateRun.newBuilder().build());
+  }
+
+  public void updateRun(String runUuid, RunStatus status, long endTime) throws Exception {
+    post("runs/update", fromMapper.makeUpdateRun(runUuid, status, endTime));
+  }
+
+  public void logParameter(String runUuid, String key, String value) throws Exception {
+    post("runs/log-parameter", fromMapper.makeLogParam(runUuid, key, value));
+  }
+
+  public void logMetric(String runUuid, String key, float value) throws Exception {
+    post("runs/log-metric", fromMapper.makeLogMetric(runUuid, key, value));
+  }
+
+  public Metric getMetric(String runUuid, String metricKey) throws Exception {
+    URIBuilder builder = new URIBuilder("metrics/get")
+      .setParameter("run_uuid", runUuid)
+      .setParameter("metric_key", metricKey);
+    return toMapper.toGetMetricResponse(httpCaller.get(builder.toString())).getMetric();
+  }
+
+  public List<Metric> getMetricHistory(String runUuid, String metricKey) throws Exception {
+    URIBuilder builder = new URIBuilder("metrics/get-history")
+      .setParameter("run_uuid", runUuid)
+      .setParameter("metric_key", metricKey);
+    return toMapper.toGetMetricHistoryResponse(httpCaller.get(builder.toString())).getMetricsList();
+  }
+
+  public ListArtifacts.Response listArtifacts(String runUuid, String path) throws Exception {
+    URIBuilder builder = new URIBuilder("artifacts/list")
+      .setParameter("run_uuid", runUuid)
+      .setParameter("path", path);
+    return toMapper.toListArtifactsResponse(httpCaller.get(builder.toString()));
+  }
+
+  public byte[] getArtifact(String runUuid, String path) throws Exception {
+    URIBuilder builder = new URIBuilder("artifacts/get")
+      .setParameter("run_uuid", runUuid)
+      .setParameter("path", path);
+    return httpCaller.getAsBytes(builder.toString());
+  }
+
+  public Optional<Experiment> getExperimentByName(String experimentName) throws Exception {
+    return listExperiments().stream().filter(e -> e.getName()
+      .equals(experimentName)).findFirst();
+  }
+
+  public String get(String path) throws Exception {
```

Comment:
Should `get` and `post` be public? If so, I'd call them doGet and doPost or something to make clear that we don't expect people to use them directly, and add Javadocs to them. Otherwise keep them private for now.